### PR TITLE
TLF-41: Add Keycloak and filevault deployment files

### PR DESCRIPTION
**Pull request by adityababumallisettiHO**
## What? 
* This work is part of ticket https://collaboration.homeoffice.gov.uk/jira/browse/TLF-50
* Update the repo with the missing deployments for filevault, keycloak-proxy
* Remove html-pdf-converter not needed
* Secret names have been updated with new names which fit for purpose
* Hof-services-configs are updated with keycloak urls

## Why? 
* Previously this is a clone of LMR. 

## How? 

* Keycloaks have been setup for notprod and prod envs
* Secrets have been added to namespace
* These deployments will now use the added secrets in namespaces

## Anything Else? (optional)
In future We will be using New keycloak-proxy image , file-vault image without vulnerabilities. Also We will be moving to Keycloak v22 from v15
## Check list

- [ ] I have reviewed my own pull request
- [ ] I have written tests (if relevant)

### CODE REVIEW

**Review by dk4g — CHANGES_REQUESTED** [review]

File: kube/file-vault/file-vault-ingress.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    {{ end }}
+  rules:
+  {{ if eq .KUBE_NAMESPACE .BRANCH_ENV }}
+  - host: fv-{{ .DRONE_SOURCE_BRANCH }}.branch.sas-notprod.homeoffice.gov.uk
+  {{ else if eq .KUBE_NAMESPACE .UAT_ENV }}
+  - host: fv-ima.uat.sas-notprod.homeoffice.gov.uk
+  {{ else if eq .KUBE_NAMESPACE .STG_ENV }}
+  - host: fv-ima.stg.sas.homeoffice.gov.uk
+  {{ else if eq .KUBE_NAMESPACE .PROD_ENV }}
+  - host: fv-ima.sas.homeoffice.gov.uk
```
[the remark this answers is not in the document]
Comment (adityababumallisettiHO) [reply kube/file-vault/file-vault-ingress.yml:37]: Thank you @dk4g . I am working on this feature branch. I have raised pull request to test the deployment on Drone.

[the remark this answers is not in the document]
Comment (adityababumallisettiHO) [reply kube/file-vault/file-vault-ingress.yml]: These file vault urls are updated now for COA

File: kube/app/deployment.yml [hunk 2]
```diff
@@ -67,48 +68,63 @@ spec:
...
+                secretKeyRef:
+                  name: redis
+                  key: pass
+            - name: USE_MOCKS
+              value: "false"
+            - name: AWS_USER
+              valueFrom:
+                secretKeyRef:
+                  name: ses
+                  key: smtp_username
```
Comment (dk4g) [inline kube/app/deployment.yml:87]: this can be removed

File: kube/app/deployment.yml [hunk 3]
```diff
@@ -67,48 +68,63 @@ spec:
...
+            - name: AWS_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: ses
+                  key: smtp_password
```
Comment (dk4g) [inline kube/app/deployment.yml:92]: `smtp_password` should be replaced with `filevault-encryption-key`

**Review by dk4g — CHANGES_REQUESTED** [review]

File: kube/app/deployment.yml (hunk 3 above)
Comment (adityababumallisettiHO) [reply kube/app/deployment.yml]: I have updated it now. Thank you

File: kube/app/deployment.yml (hunk 2 above)
Comment (adityababumallisettiHO) [reply kube/app/deployment.yml]: I have removed these secret key references which aren't in the new secret created

File: .drone.yml [hunk 4]
```diff
@@ -335,7 +335,7 @@ steps:
     image: 340268328991.dkr.ecr.eu-west-2.amazonaws.com/acp/trivy/client:latest
     pull: always
     environment:
-        IMAGE_NAME: coa:${DRONE_COMMIT_SHA}
+        IMAGE_NAME: :${DRONE_COMMIT_SHA}
```
Comment (dk4g) [inline .drone.yml:338]: Not sure how the `IMAGE_NAME` is used later, but there is an extra colon, before there was a custom name `coa`, perhaps the `$IMAGE_REPO` is missing?

**Review by dk4g — APPROVED** [review]

**Pull request merged**